fuzzbench/run_experiment.py

```diff
@@ -13,21 +13,25 @@
 # limitations under the License.
 """Runs a FuzzBench experiment."""
 
+import os
 import time
 
 import redis
 import rq
 
+from common import config_utils, yaml_utils
 from fuzzbench import fake_jobs
 
 
-def run_experiment():
+def run_experiment(config):
     """Main experiment logic."""
     print('Initializing the job queue.')
     queue = rq.Queue()
     jobs = []
     for i in range(6):
         jobs.append(queue.enqueue(fake_jobs.build_image, 'something-%d' % i))
+    if config.get('end_to_end_test', False):
+        jobs.append(queue.enqueue(fake_jobs.fake_job, job_id='e2e-test'))
```

Comment:
why are we doing this? can we build an image instead? this code shouldn't behave differently when we're running an e2e test. the e2e test is for validating the normal behavior. only the configuration should be different (see other comment)

Reply:
> why are we doing this? can we build an image instead? this code shouldn't behave differently when we're running an e2e test. the e2e test is for validating the normal behavior. only the configuration should be different (see other comment)

I think the main point lying in that, do we need to test the end-to-end testing is using the desired config file. This comment argument point is based on we actually don't need this test, which means we already know the end-to-end test will use the desired config file.

I think this test included in the end-to-end test is needed to check. This is more like, we need to first sure run on an experiment we desire and then we check experimental behavior.

Otherwise, how to know the modification in `compose/e2e-test.yaml` is correct? wdyt

Reply:
yes tests thing cannot be in main code.

Reply:
> yes tests thing cannot be in main code.

For end-to-end test, since we cannot monitor the experiment but only check the outcome of the experiment, thus I need to have some specific outcome for this experiment. Otherwise, we cannot check whether it is using the desired config file if there is no special thing happen.

I think currently we should first try to agree or disagree on, whether we should test whether this e2e test is using the desired config file. Then, we can discuss how to implement this goal. 

wdyt?